chargeOclock/Source/Models/Client.swift

```diff
@@ -15,10 +15,15 @@ protocol ClientSettable {
 class Client {
 
 	var id: Int = 0
-	var name: String = ""
+	var isDirty: Bool = false
 
-	init(id: Int, name: String) {
-		self.id = id
+	var name: String = "" {
+		didSet {
+			isDirty = true
+		}
+	}
+
+	init(name: String) {
 		self.name = name
```

Comment:
so, the logic is this:

new client:  the object is created with a specific name (constructor): set the name, set the isDirty flag (since this object hasn't been stored yet), set the id to nil

save client: check if the isDirty flag is set, if not exit. check if we have an id, if so, this is an existing client (so update), if not, it's a new client (so create). clear the isDirty flag (if no error occurred, of course), set the new id, if this was a create call.

read client (from json): just initialize the values from the json, clear the isDirty flag.

any change to any property (only name in this case): set the isDirty flag.

setting the isDirty flag means setting it to true, clearing the flag means setting it to false.